### DISCUSSION TITLE
Use epgu in TB3 legacy mode, epgu has no output, Although I can see my GPU in the PCI list

### DISCUSSION THREAD

**Discussion by AlferWu** [osy86/HaC-Mini#580] [Q&A]
**PLEASE READ THE FOLLOWING**
* Make sure you installed the latest version of HaC Mini
* Follow **all** the [troubleshooting steps](https://osy.gitbook.io/hac-mini-guide/post-installation/support)
* Check the open issues and make sure it is not already reported
* Delete this section from your report and continue

**Describe the bug**
Use epgu in TB3 legacy mode, epgu has no output, Although I can see my GPU in the PCI list
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/16063974/117831694-66dd6500-b2a7-11eb-83fe-2efae7da79b5.png"> (**my rx590 showed driver uninstalled)**

When I use the native mode the egpu works fine,  but , Because I have a dual system(MAC/ Windows ), it still cannot be displayed on the system picker view, and it will only work  after system fully booted.

**Expected behavior**
egpu work under legacy mode ,and it better to works under system picker.

**Configuration**
- HaC Mini version [e.g: 2.0]: 2.17
- OSX version [e.g: 10.14.6]: 11.0.1
- NUC model [e.g: NUC8i7HNK]: NUC8I7HVK
- Boot SSD model and protocol [e.g: Samsung EVO 860 1TB, SATA]: C2000PRO 1TB NVME
- Second SSD model and protocol (if installed) [e.g: HP EX920 1TB, NVMe]: 
- First DIMM model [e.g: Patriot DDR4-2666 16GB]: Corsair Gaming DDR4 2400 16GB
- Second DIMM model (if installed): Corsair Gaming DDR4 2400 16GB
- Display port(s) used [e.g: USB-C, mini DP]: egpu
- Display resolution(s) [e.g: 4K, 5K]: 4k
- Any TB3 devices: Mantiz Venus eGPU 
- Any USB devices: keyboard
- Any Wi-Fi/Bluetooth devices [e.g:DW1820A, BCM94360CS2, Intel 8265]: BCM94360CS2

**Modifications**
List any mods to `config.plist`, any additional kexts, anything installed to `/Library/Extensions` and `/System/Library/Extensions`, any additional patches, etc. Please attach your `config.plist` if you made any modifications.

**Panic log**
Include a screenshot of the panic message if applicable.

**Comment by osy** [comment]
The HaC mini project does not support eGPU in legacy mode (it's called legacy mode because it's not supported). However, I think there has been people who've gotten in working before who may have some insight. In legacy mode, the PCIe controller is seen by macOS as a regular PCIe slot with hotplug capabilities. Therefore it does not know anything about the thunderbolt controller.

One alternative is that in native mode, if you boot up into macOS first and restart into windows all your connected devices will still be seen. As long as the link is established you can continue to use the device. However, if you sleep or unplug the device it will not work until you boot macOS again.
